### PR TITLE
[breaking] consistent casing in `goto`

### LINKED ISSUE #7460: Make `goto` options casing more consistent?
### Describe the problem

`goto` has this signature:
```ts
function goto(
  url: string | URL,
  opts?: {
    replaceState?: boolean;
    noscroll?: boolean;
    keepfocus?: boolean;
    state?: any;
  }
): Promise<void>;
```

`replaceState` is camel-cased while `noscroll` and `keepfocus` are not.

### Describe the proposed solution

Rename:
- `noscroll` -> `noScroll`
- `keepfocus` -> `keepFocus`

(This also would make spell checkers happier.)

### Alternatives considered

_No response_

### Importance

nice to have

### Additional Information

_No response_

### PR DESCRIPTION
**Pull request by aantusahaa**
Closes #7460

### Please don't delete this checklist! Before submitting the PR, please make sure you do the following:
- [x] It's really useful if your PR references an issue where it is discussed ahead of time. In many cases, features are absent for a reason. For large changes, please create an RFC: https://github.com/sveltejs/rfcs
- [x] This message body should clearly illustrate what problems it solves.
- [ ] Ideally, include a test that fails without this PR but passes with it.

### Tests
- [ ] Run the tests with `pnpm test` and lint the project with `pnpm lint` and `pnpm check`

### Changesets
- [ ] If your PR makes a change that should be noted in one or more packages' changelogs, generate a changeset by running `pnpm changeset` and following the prompts. All changesets should be `patch` until SvelteKit 1.0

### CODE REVIEW

**Comment by dummdidumm** [comment]
Thank you! Could you add checks to the public API call sites to check if people are calling it with the old casing, and throwing an error in that case? Something along the lines of "X has been renamed to Y (note the difference in casing)".

**Comment by aantusahaa** [comment]
`keepfocus` could also be renamed for `update` and `navigate` methods as well. Should that be done?

**Comment by dummdidumm** [comment]
Good question. It would safe a couple of characters in the final output because you can then use shorthands, but it conflicts with the internal coding style. I'm a little towards changing them as well, I'd wait for other opinions before making a decision.

**Comment by dummdidumm** [comment]
Yes, this is what I meant, although it's probably better to add this to the `goto` method on line ~1152 (`goto: (href, opts = ..`) since that's the public interface

File: packages/kit/src/runtime/client/client.js [hunk 1]
```diff
@@ -1149,7 +1149,21 @@ export function create_client({ target, base, trailing_slash }) {
...
 
-		goto: (href, opts = {}) => goto(href, opts, []),
+		goto: (href, opts = {}) => {
+			if (opts.hasOwnProperty('keepfocus')) {
+				throw new Error(
+					'`keepfocus` has been renamed to `keepFocus` (note the difference in casing)'
+				);
+			}
+
+			if (opts.hasOwnProperty('noscroll')) {
```
Comment (Rich-Harris) [inline packages/kit/src/runtime/client/client.js:1159]: `property in opts` is equivalent for all intents and purposes but is more byte-efficient

```suggestion
			// TODO remove for 1.0
			if ('keepfocus' in opts) {
				throw new Error(
					'`keepfocus` has been renamed to `keepFocus` (note the difference in casing)'
				);
			}

			if ('noscroll' in opts) {
```

**Comment by Rich-Harris** [comment]
thank you!

**Pull request merged**